pkg/m3admin/placement/placement.go

```diff
@@ -0,0 +1,197 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package placement
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/m3db/m3cluster/generated/proto/placementpb"
+	"github.com/m3db/m3db-operator/pkg/m3admin"
+
+	"github.com/gogo/protobuf/jsonpb"
+	retryhttp "github.com/hashicorp/go-retryablehttp"
+	plc "github.com/m3db/m3/src/query/api/v1/handler/placement"
+	"github.com/m3db/m3/src/query/generated/proto/admin"
+	"github.com/m3db/m3/src/query/util/logging"
+	"go.uber.org/zap"
+)
+
+// ErrPlacementNotFound indicates that the placement is not found
+var ErrPlacementNotFound = errors.New("placement not found")
+
+const (
+	_defaultPlacementURI = "/api/v1/placement"
+)
+
+type placement struct {
+	servicePort   int
+	serviceName   string
+	serviceDomain string
+	client        *retryhttp.Client
+	logger        *zap.Logger
+}
+
+// Option provides an interface that can be used for setter options with the
+// constructor
+type Option interface {
+	execute(*placement) error
+}
+
+type optionFn func(p *placement) error
+
+func (fn optionFn) execute(p *placement) error {
+	return fn(p)
+}
+
+// WithPort is a setter to override the default port of 7201
+func WithPort(port int) Option {
+	return optionFn(func(p *placement) error {
+		p.servicePort = port
+		return nil
+	})
+}
+
+// WithName is a setter to override the default name of m3coordinator
+func WithName(name string) Option {
+	return optionFn(func(p *placement) error {
+		p.serviceName = name
+		return nil
+	})
+}
+
+// WithDomain is a setter to override the default name of default
+func WithDomain(name string) Option {
+	return optionFn(func(p *placement) error {
+		p.serviceDomain = name
+		return nil
+	})
+}
+
+// WithLogger is a setter to override the default logger
+func WithLogger(logger *zap.Logger) Option {
+	return optionFn(func(p *placement) error {
+		p.logger = logger
+		return nil
+	})
+}
+
+// New is the constructor the Placement interface
+func New(opts ...Option) (Placement, error) {
+	logging.InitWithCores(nil)
```

Comment:
why initialise logging here? 

would be better to use a https://godoc.org/go.uber.org/zap#NewNop here by default and let users override with the option if so desired

Reply:
Great call! Done.